File: Training/Classic Puzzle - Easy/Horse_Racing_Duals.cpp
/**
 * @file Horse_Racing_Duals.cpp
 * @author Ladjouzi Rachid (RashidLadj@gmail.com)
 * @brief 
 * @version 0.1
 * @date 2021-10-11
 * 
 * @copyright Copyright (c) 2021
 */

#include <iostream>
#include <vector>
#include <algorithm>

using namespace std;


int main(){
    int N;   cin >> N;   cin.ignore();

    vector<int> mySet;
    
    for (size_t i = 0; i < N; i++) {
        int Pi; cin >> Pi; cin.ignore();
        mySet.push_back(Pi);
    }

    std::sort(mySet.begin(), mySet.end());

    int diff = -1;
    for (size_t i = 1; i < mySet.size(); ++i){
        int d = abs(mySet[i] - mySet[i-1]);
        if (d < diff || diff == -1)
            diff = d;
    }

    cout << diff << endl;
}

File: Training/Classic Puzzle - Easy/Horse_Racing_Duals.py
"""
 @file Horse_Racing_Duals.py
 @author Ladjouzi Rachid (RashidLadj@gmail.com)
 @brief 
 @version 0.1
 @date 2021-10-12
 
 @copyright Copyright (c) 2021
"""

n = int(input())
myList = []
for i in range(n):
    myList.append(int(input()))

myList = sorted(myList)

diff = -1
for i in range (1, n):
    d = abs(myList[i] - myList[i-1])
    if d < diff or diff == -1:
        diff = d
print(diff)

